### PR TITLE
Remove TKM string code

### PR DESCRIPTION
**Pull request by frosch123**
## Motivation / Problem

#11341 added a `{TKM "calendar" "wallclock"}` string control code:
* It would insert the strings `calendar` or `wallclock` into the string depending on a game setting value.
* This command is not supported by other tools (eints).
* The "calender" and "wallclock" texts do now allow any cases/genders/plurals, which may be required in certain languages and/or future strings.
* In the case of vehicle service intervals: there are actually three cases ("days", "minutes", "percent"). TKM cannot handle this.
* Game settings are rather complex, and should be handled in the code. Templating the strings depending on settings moves too much magic into the strings.

## Description

This PR removes usage and support for TKM:
* Strings are duplicated with slight alterations. This pattern was common before.
* The `UsingWallclockUnits` is called from the individual GUI code to choose string variants.
* `UsingWallclockUnits` must know, whether it should check the current-game or new-game settings. The invidual GUI code knows this better than the general string templating code.
* `UsingWallclockUnits` is still called with the magic `_game_mode == GM_MENU` in the settings GUI. I am not aware of any case, where this assumption is false.
* Settings now support callbacks to supply dynamic title, help and value strings.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/graph_gui.cpp [hunk 1]
```diff
@@ -492,7 +492,14 @@ struct BaseGraphWindow : Window {
 		/* Initialise the dataset */
 		this->UpdateStatistics(true);
 
-		this->InitNested(number);
+		this->CreateNestedTree();
+
+		auto *wid = this->GetWidget<NWidgetCore>(WID_GRAPH_FOOTER);
+		if (wid != nullptr && TimerGameEconomy::UsingWallclockUnits()) {
+			wid->SetDataTip(STR_GRAPH_LAST_72_MINUTES_TIME_LABEL, STR_NULL);
```
Comment (rubidium42) [inline src/graph_gui.cpp:499]: Should we, for consistency, also have a string when it's not using wall clock units?
This PR keeps it as is, but it feels weird only having a tooltip some of the times. Possibly not for this PR though.

File: src/settings_internal.h [hunk 2]
```diff
@@ -142,6 +141,10 @@ struct SettingDesc {
 
 /** Base integer type, including boolean, settings. Only these are shown in the settings UI. */
 struct IntSettingDesc : SettingDesc {
+	typedef StringID GetTitleCallback(const IntSettingDesc& sd);
+	typedef StringID GetHelpCallback(const IntSettingDesc& sd);
+	typedef void SetValueDParamsCallback(const IntSettingDesc& sd, uint first_param, int32_t value);
```
Comment (rubidium42) [inline src/settings_internal.h]:
```suggestion
	typedef StringID GetTitleCallback(const IntSettingDesc &sd);
	typedef StringID GetHelpCallback(const IntSettingDesc &sd);
	typedef void SetValueDParamsCallback(const IntSettingDesc &sd, uint first_param, int32_t value);
```

File: src/settings_table.cpp [hunk 3]
```diff
@@ -81,6 +81,40 @@ SettingTable _win32_settings{ _win32_settings_table };
 
 /* Begin - Callback Functions for the various settings. */
 
+/** Switch setting title depending on wallclock setting */
+static StringID SettingTitleWallclock(const IntSettingDesc& sd)
```
Comment (rubidium42) [inline src/settings_table.cpp]:
```suggestion
static StringID SettingTitleWallclock(const IntSettingDesc &sd)
```

File: src/settings_table.cpp [hunk 4]
```diff
@@ -81,6 +81,40 @@ SettingTable _win32_settings{ _win32_settings_table };
...
+{
+	return TimerGameEconomy::UsingWallclockUnits(_game_mode == GM_MENU) ? sd.str + 1 : sd.str;
+}
+
+/** Switch setting help depending on wallclock setting */
+static StringID SettingHelpWallclock(const IntSettingDesc& sd)
```
Comment (rubidium42) [inline src/settings_table.cpp]:
```suggestion
static StringID SettingHelpWallclock(const IntSettingDesc &sd)
```

File: src/settings_table.cpp [hunk 5]
```diff
@@ -81,6 +81,40 @@ SettingTable _win32_settings{ _win32_settings_table };
...
+		case 2: val = STR_CONFIG_SETTING_LOCALISATION_UNITS_VELOCITY_SI; break;
+		case 3: val = TimerGameEconomy::UsingWallclockUnits(_game_mode == GM_MENU) ? STR_CONFIG_SETTING_LOCALISATION_UNITS_VELOCITY_GAMEUNITS_SECS : STR_CONFIG_SETTING_LOCALISATION_UNITS_VELOCITY_GAMEUNITS_DAYS; break;
+		case 4: val = STR_CONFIG_SETTING_LOCALISATION_UNITS_VELOCITY_KNOTS; break;
+		default: NOT_REACHED();
+	}
+	SetDParam(first_param, val);
+}
+
+/** A negative value has another string (the one after "strval"). */
+static void SettingsValueAbsolute(const IntSettingDesc& sd, uint first_param, int32_t value)
```
Comment (rubidium42) [inline src/settings_table.cpp]:
```suggestion
static void SettingsValueAbsolute(const IntSettingDesc &sd, uint first_param, int32_t value)
```

File: src/table/settings.h.preamble [hunk 6]
```diff
@@ -10,6 +10,8 @@
 /* Callback function used in _settings[] as well as _company_settings[] */
 static size_t ConvertLandscape(const char *value);
 
+static StringID SettingTitleWallclock(const IntSettingDesc& sd);
+static StringID SettingHelpWallclock(const IntSettingDesc& sd);
```
Comment (rubidium42) [inline src/table/settings.h.preamble]:
```suggestion
static StringID SettingTitleWallclock(const IntSettingDesc &sd);
static StringID SettingHelpWallclock(const IntSettingDesc &sd);
```

File: src/table/settings/company_settings.ini [hunk 7]
```diff
@@ -11,13 +11,14 @@
 static void UpdateAllServiceInterval(int32_t new_value);
 static bool CanUpdateServiceInterval(VehicleType type, int32_t &new_value);
 static void UpdateServiceInterval(VehicleType type, int32_t new_value);
+static void SettingsValueAbsolute(const IntSettingDesc& sd, uint first_param, int32_t value);
```
Comment (rubidium42) [inline src/table/settings/company_settings.ini]:
```suggestion
static void UpdateServiceInterval(VehicleType type, int32_t new_value);
static void SettingsValueAbsolute(const IntSettingDesc &sd, uint first_param, int32_t value);
```

File: src/table/settings/locale_settings.ini [hunk 8]
```diff
@@ -8,6 +8,8 @@
 ; in the savegame PATS chunk. These settings are not sync'd over the network.
 
 [pre-amble]
+static void SettingsValueVelocityUnit(const IntSettingDesc& sd, uint first_param, int32_t value);
```
Comment (rubidium42) [inline src/table/settings/locale_settings.ini]:
```suggestion
static void SettingsValueVelocityUnit(const IntSettingDesc &sd, uint first_param, int32_t value);
```

**Review by rubidium42 — COMMENTED** [review]
Looks okay to me, barring some minor coding style nitpicks.

File: src/graph_gui.cpp (hunk 1 above)
Comment (frosch123) [reply src/graph_gui.cpp:499]: There technically is, but different:
* Calendar: show year numbers along the X axis.
* Wallclock: there are no labels on the X axis, only this general text "72 minutes in total".

**Review by rubidium42 — APPROVED** [review]

**Pull request merged**